### PR TITLE
Fix incorrect skaffold init output in multi-module projects

### LINKED ISSUE #2262: _skaffold-init task multimodule logic issues
The issues with the current logic (at least in the maven plugin, gradle task might have the same issues): 

- when a submodule has a different parent specified (e.g. spring boot starter pom) then our current logic fails to set the project id as the parent comes back as null in the InitMojo - this creates issues in multimodule projects (e.g. the Cloud Code java guestbook sample) because the listed projects don't have anything in them
- I'm not sure how frequent nested modules are but currently we don't handle those well either
- we are using the `artifactId` instead of the module name to return a `project` which is a problem as that is interpreted on the skaffold side as a directory

I won't have time to fully implement the thing with tests and everything, but as I played around with the code this seemed to do the job altogether: 

```java
@Mojo(name = InitMojo.GOAL_NAME, requiresDependencyCollection = ResolutionScope.NONE)
public class InitMojo extends JibPluginConfiguration {

  @VisibleForTesting static final String GOAL_NAME = "_skaffold-init";

  @Override
  public void execute() throws MojoExecutionException {
    checkJibVersion();
    MavenProject project = getProject();
    MavenProject topLevelProject = getSession().getTopLevelProject();
    if (!project.equals(topLevelProject)) {
      return;
    }

    if (isParentProject(project)) {
      Set<MavenProject> projects = collectLeafProjects(project);
      for (MavenProject p : projects) {
        print(p, true);
      }
    } else {
      print(project, false);
    }
  }

  private Set<MavenProject> collectLeafProjects(MavenProject project) {
    Set<MavenProject> projects = new HashSet<>();
    for (MavenProject p : project.getCollectedProjects()) {
      if (isParentProject(p)) {
        projects.addAll(collectLeafProjects(p));
      } else {
        projects.add(p);
      }
    }
    return projects;
  }

  private boolean isParentProject(MavenProject project) {
    return project.getPackaging().equals("pom") && project.getModules().size() > 0;
  }

  private void print(MavenProject project, boolean setProject) throws MojoExecutionException {
    SkaffoldInitOutput skaffoldInitOutput = new SkaffoldInitOutput();
    skaffoldInitOutput.setImage(getTargetImage());
    if (setProject) {
      skaffoldInitOutput.setProject(project.getGroupId() + ":" + project.getArtifactId());
    }
    System.out.println();
    System.out.println("BEGIN JIB JSON");
    try {
      System.out.println(skaffoldInitOutput.getJsonString());
    } catch (IOException ex) {
      throw new MojoExecutionException(ex.getMessage(), ex);
    }
  }
}

```

### PR DESCRIPTION
**Pull request by TadCordle**
Fixes #2262.

### CODE REVIEW

**Comment by TadCordle** [comment]
Actually, did some manual tests with this after putting the PR up, and realized the target image will no longer print out for submodules.

**Comment by loosebazooka** [comment]
how are we invoking init? What happens if the project doesn't have jib on it?

File: jib-maven-plugin/src/test/resources/maven/projects/spring-boot-multi/pom.xml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+       <id>snapshots-repo</id>
+       <url>https://oss.sonatype.org/content/repositories/snapshots</url>
+       <releases>
+         <enabled>false</enabled>
+       </releases>
+       <snapshots>
+         <enabled>true</enabled>
+       </snapshots>
+     </repository>
+   </repositories>
```
Comment (chanseokoh) [inline jib-maven-plugin/src/test/resources/maven/projects/spring-boot-multi/pom.xml]: This is probably not needed for this test.

File: jib-maven-plugin/src/test/resources/maven/projects/spring-boot-multi/service-1/pom.xml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    <pluginManagement>
+      <plugins>
+        <plugin>
+          <groupId>com.google.cloud.tools</groupId>
+          <artifactId>jib-maven-plugin</artifactId>
+          <version>@@PluginVersion@@</version>
+        </plugin>
+      </plugins>
+    </pluginManagement>
+  </build>
```
Comment (chanseokoh) [inline jib-maven-plugin/src/test/resources/maven/projects/spring-boot-multi/service-1/pom.xml]: I bet you've copied this whole setup from another directory, but isn't this redundant?

File: jib-maven-plugin/src/test/resources/maven/projects/spring-boot-multi/service-2/pom.xml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  <artifactId>service-2</artifactId>
+  <version>1.0.0.TEST-SNAPSHOT</version>
+
+  <parent>
+    <groupId>org.springframework.boot</groupId>
+    <artifactId>spring-boot-starter-parent</artifactId>
+    <version>2.1.6.RELEASE</version>
+  </parent>
+
+  <build>
```
Comment (chanseokoh) [inline jib-maven-plugin/src/test/resources/maven/projects/spring-boot-multi/service-2/pom.xml]: ditto

File: jib-maven-plugin/src/test/resources/maven/projects/spring-boot-multi/service-1/pom.xml (hunk 2 above)
Comment (loosebazooka) [reply jib-maven-plugin/src/test/resources/maven/projects/spring-boot-multi/service-1/pom.xml]: is this maybe an indication that we might need to do some test cleanup?

Comment (loosebazooka) [reply jib-maven-plugin/src/test/resources/maven/projects/spring-boot-multi/service-1/pom.xml]: oops, just found https://github.com/GoogleContainerTools/jib/issues/2150 again.

Comment (chanseokoh) [reply jib-maven-plugin/src/test/resources/maven/projects/spring-boot-multi/service-1/pom.xml]: Ah, then I think the `pluginManagement` in the root `pom.xml` is unnecessary?

Comment (TadCordle) [reply jib-maven-plugin/src/test/resources/maven/projects/spring-boot-multi/service-1/pom.xml]: I think I want to keep it to ensure that the output isn't printed on the root project, even though the Jib plugin is applied?

**Comment by TadCordle** [comment]
@balopat would you mind checking out this branch and seeing if it fixes the issues you were seeing? I took a different approach than you suggested, so it's possible I may have missed an edge case, but from my manual testing I haven't noticed any issues yet.

**Comment by balopat** [comment]
Looks good to me! Thank you for fixing it @TadCordle!!

**Pull request merged**